tests/Core.nut

```diff
@@ -0,0 +1,422 @@
+    // Should return true, if the test was successful, false otherwise.
+    // If not specified, then by default the test will be successful (or not, considering the statuscode).
+    "callbackVerify": null,
+    // Specifies how many times program should send requests to the server.
+    "numberOfRequests": 1,
+    // Middleware or array of middlewares for Rocky.use
+    "mwApp": [],
+    // Middleware or array of middlewares for Rocky.Route.use
+    "mw": [],
+    // Specifies handlers for Rocky.authorize|Rocky.onUnauthorized|Rocky.onTimeout|Rocky.onNotFound|Rocky.onException
+    "onAuthorizeApp": null,
+    "onUnauthorizedApp": null,
+    "onTimeoutApp": null,
+    "onNotFoundApp": null,
+    "onExceptionApp": null,
+    // Specifies handlers for Rocky.Route.authorize|Rocky.Route.onUnauthorized|Rocky.Route.onTimeout|Rocky.Route.onException
+    "onAuthorizeRoute": null,
+    "onUnauthorizedRoute": null,
+    "onTimeoutRoute": null,
+    "onExceptionRoute": null
+};
+
+// createTest
+// Create Promise for testing Rocky.
+// All customization is carried out through a single parameter - table. Table's params described above.
+// Scenario of the method:
+// 1) Apply default values to 'params' if undefined
+// 2) Create Rocky instance
+// 3) Set Rocky's handlers from 'params' if defined (Rocky.authorize, Rocky.onUnauthorized, Rocky.onTimeout, Rocky.onNotFound, Rocky.onException)
+// 4) Set Rocky's middlewares from 'params' if defined (Rocky.use)
+// 5) Create handler for incoming requests (Rocky.Route)
+// 6) Set Rocky.Route's handlers from 'params' if defined (Rocky.Route.authorize, Rocky.Route.onUnauthorized, Rocky.Route.onTimeout, Rocky.Route.onException)
+// 7) Set Rocky.Route's middlewares from 'params' if defined (Rocky.Route.use)
+// 8) Send one (or more) request(s) and wait for Rocky's response.
+//    When get response, verify statuscode and call callback function for additional verify (if defined). If everything ok, then test passed, otherwise failed.
+//    If more than one requests send, then test will be passed only when all responses verifications passed.
+// 
+// @param {table} params
+// @param {string} expect - Expected completion of the test (success|fail)
+// @return {Promise}
+function createTest(params = {}, expect = "success") {
+    return Promise(function(resolve, reject) {
+        local success = function() {
+            if (expect == "fail") {
+                reject("createTest test was resolved, but 'fail' expected");
+            } else {
+                resolve();
+            }
+        };
+        local fail = function(reason = null) {
+            if (expect == "fail") {
+                resolve();
+            } else {
+                reject(reason);
+            }
+        };
+        local app;
+        local route;
+        // Setup default values
+        params.setdelegate(defaultParams);

```

Comment:
Delegate is used to set the default values. If there is no parameter in the params table, it will be taken from the defaultParams table.